privacyidea/lib/utils/__init__.py

```diff
@@ -888,6 +888,34 @@ def compare_value_value(value1, comparator, value2):
     raise Exception("Unknown comparator: {0!s}".format(comparator))
 
 
+def compare_generic_condition(cond, key_method, warning):
+    """
+    Compares a condition like "tokeninfoattribute == value".
+    It uses the "key_method" to determine the value of "tokeninfoattribute".
+
+    If the value does not match, it returns False.
+
+    :param cond: A condition containing a comparator like "==", ">", "<"
+    :param key_method: A function call, that get the value from the key
+    :param warning: A warnnig message to be written to the log file.
+    :return: True of False
+    """
+    key = value = None
+    for comparator in ["==", ">", "<", None]:
```

Comment:
Using the comparator `None` will lead to an error `compare_value_value()` when the condition is something like "tokeninfoattribute value" (Who knows what users put in the inputs).

Reply:
I needed "None" as breakpoint. If you have a better suggestion, shoot! :-)

Reply:
I actually think, this might be a quite interesting result.
The problem is, what would s.o. expect if he enters a wrong condition. Should it be false or true?
So raising an exception might even be nicer, since the admin actually realizes the problem.
I can totally live with that.

Reply:
done in 848e1a32

Reply:
I would agree, an exception hints at a wrong condition.
I thought first the support of <= or >= would be nice, but in principle nobody would actually need those here. Just sharing the thought.